### PR TITLE
feat(traffic_light_classifier): update traffic_light_classifier and traffic_light_fine_detector

### PR DESCRIPTION
**Pull request by Mingyu1991**
## Description

- Updated tensorrt_common, tensorrt_yolox and tensorrt_classifier by @dan-dnn 
- Updated traffic_light_classifier and traffic_light_fine_detector using the above packages.

## Tests performed

Nothing.

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tensorrt_common/include/tensorrt_common/tensorrt_common.hpp [hunk 1]
```diff
@@ -201,20 +145,28 @@ class TrtCommon  // NOLINT
...
   // profiler for whole model
-  SimpleProfiler host_profiler_;
-
-  std::unique_ptr<const BuildConfig> build_config_;
+  SimpleProfiler m_host_profiler;
+  // type for calibration
+  nvinfer1::CalibrationAlgoType m_calibType;
+  // flg for dla
+  int m_dla;
+  // flg for partial quanitzation in first layer
```
Comment (yukkysaito) [inline common/tensorrt_common/include/tensorrt_common/tensorrt_common.hpp:162]:
```suggestion
  // flg for partial quantization in first layer
```

File: common/tensorrt_common/include/tensorrt_common/tensorrt_common.hpp [hunk 2]
```diff
@@ -201,20 +145,28 @@ class TrtCommon  // NOLINT
...
+  bool m_first;
+  // flg for partial quanitzation in last layer
```
Comment (yukkysaito) [inline common/tensorrt_common/include/tensorrt_common/tensorrt_common.hpp:164]:
```suggestion
  // flg for partial quantization in last layer
```

File: common/tensorrt_common/include/tensorrt_common/tensorrt_common.hpp (hunk 1 above)
Comment (Mingyu1991) [reply common/tensorrt_common/include/tensorrt_common/tensorrt_common.hpp]: fixed in commit [d1a2ec5](https://github.com/autowarefoundation/autoware.universe/pull/4125/commits/d1a2ec58dee7ade4b4b0cf2c582817e88c1122dd)

File: common/tensorrt_common/include/tensorrt_common/tensorrt_common.hpp (hunk 2 above)
Comment (Mingyu1991) [reply common/tensorrt_common/include/tensorrt_common/tensorrt_common.hpp]: fixed in commit [d1a2ec5](https://github.com/autowarefoundation/autoware.universe/pull/4125/commits/d1a2ec58dee7ade4b4b0cf2c582817e88c1122dd)

File: perception/tensorrt_yolox/src/tensorrt_yolox_node.cpp [hunk 3]
```diff
@@ -199,8 +207,6 @@ void TrtYoloXNode::replaceLabelMap()
     auto & label = label_map_[i];
     if (label == "PERSON") {
       label = "PEDESTRIAN";
-    } else if (label == "MOTORBIKE") {
-      label = "MOTORCYCLE";
```
Comment (miursh) [inline perception/tensorrt_yolox/src/tensorrt_yolox_node.cpp:203]: [must] Need to keep these lines

**Review by miursh — COMMENTED** [review]
In this PR, there are lots of change in tensorrt_yolox, but I don't think all of them are necessary. 
It appears some of the codes have reverted back to previous state.

**Review by miursh — APPROVED** [review]
LGTM

**Comment by miursh** [comment]
Screenshot from 2023-07-08 11-20-34
I have confirmed that it works as intended.

**Pull request merged**